pkg/controller/networkpolicy/networkpolicy_controller_perf_test.go

```diff
@@ -127,18 +127,60 @@ func TestInitXLargeScaleWithSmallNamespaces(t *testing.T) {
 	testComputeNetworkPolicy(t, 10*time.Second, namespaces, networkPolicies, pods)
 }
 
-func testComputeNetworkPolicy(t *testing.T, maxExecutionTime time.Duration, namespaces []*v1.Namespace, networkPolicies []*networkingv1.NetworkPolicy, pods []*v1.Pod) {
-	objs := make([]runtime.Object, 0, len(namespaces)+len(networkPolicies)+len(pods))
-	for i := range namespaces {
-		objs = append(objs, namespaces[i])
-	}
-	for i := range networkPolicies {
-		objs = append(objs, networkPolicies[i])
-	}
-	for i := range pods {
-		objs = append(objs, pods[i])
+//TODO: standardize performance testing across tests in controlled environment
+/*
+TestInitXLargeScaleWithOneNamespaces tests the execution time and the memory usage of computing a scale
+of 1 Namespaces, 10k NetworkPolicies, 10k Pods where each network policy selects each pod (applied + ingress).
+
+NAMESPACES   PODS    NETWORK-POLICIES    TIME(s)    MEMORY(M)    EXECUTIONS    EVENTS(ag, atg, np)
+1            10000   10000               10.66       1157         30380         20368 5 20368
+
+The metrics are not accurate under the race detector, and will be skipped when testing with "-race".
+*/
+func TestInitXLargeScaleWithOneNamespace(t *testing.T) {
```

Comment:
Is `TestInitXLargeScaleWithOneNamespace` used anywhere? I couldn't find a reference to it.

Also did you try reusing some code between the existing "perf" tests and your new "perf" tests? I think you did since I can see you are using `getXObjects`.

Reply:
TestInitXLargeScaleWithOneNamespace is a test I added to the controller which measures controller initialization performance for a new relevant case (relevant because of the new indexing), testing performance when a large number of policies are applied to the each pod. "OneNamespace" is a condition which yields this setup (large number of policies per pod).

Reply:
Yes, I tried reusing some code between the files, getXObjects is very helpful. I can't find any other clear ways in which I can reuse code (other than sharing policies, which at this scale I think would only add complexity). If you see any other ways in which I can reuse code I can make the changes.

Reply:
Sounds good, we'll see if someone has suggestions once we submit the PR upstream